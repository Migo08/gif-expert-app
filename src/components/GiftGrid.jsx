import { useEffect, useState } from "react";
import { useFetchGifs } from "../hooks/useFetchGifs";
import { GiftItem } from "./GiftItem";
import PropTypes from 'prop-types';

export const GiftGrid = ({ category }) => {

    const { images, isLoading } = useFetchGifs( category );
    return (
        <>
            <h3>{ category }</h3>
            {
                isLoading ? ( <h2>Cargando...</h2> ) : null
            }

            {
                isLoading && ( <h2>Cargando 2...</h2> )
            }
            
            <div className="card-grid" aria-label="cardgrid">
                {   
                    images.map( image => (
                        <GiftItem 
                            key={image.id} 
                            { ...image } 
                        />
                    ))
                }
            </div>
        </>
        
    )
}

GiftGrid.propTypes = {
    category: PropTypes.string.isRequired
}