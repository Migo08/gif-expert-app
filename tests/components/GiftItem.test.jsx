import { render, screen } from "@testing-library/react";
import { GiftItem } from "../../src/components/GiftItem";



const title = 'Prueba';
const url   = 'https://media1.giphy.com/media/gIIIX3GbqJmkzxn9Mk/giphy.gif?cid=3285e6afm7o8ywcrpqe3md5oumk3amjk2b14f5qa7sfp77iq&rid=giphy.gif&ct=g';

describe('Pruebas en <GiftItem />', () => { 
    test('Debe hacer match con el snapshop', () => { 
        const { container } = render(<GiftItem title={ title } url={ url } />);
        expect( container ).toMatchSnapshot();
    });

    test('Debe mostrar imagen con url y alt indicado', () => { 
        render( <GiftItem title={ title } url={ url } /> );
        const { src, alt } = screen.getByRole('img');
        expect( src ).toBe( url);
        expect( alt ).toBe( title );
     });

     test('Debe de mostrar el tÃ­tulo en el componente ', () => { 
        render( <GiftItem title={ title } url={ url } /> );
        expect( screen.getByText( title ) ).toBeTruthy();
     });
});