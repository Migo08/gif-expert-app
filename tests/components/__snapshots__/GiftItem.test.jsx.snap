// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GiftItem /> Debe hacer match con el snapshop 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Prueba"
      src="https://media1.giphy.com/media/gIIIX3GbqJmkzxn9Mk/giphy.gif?cid=3285e6afm7o8ywcrpqe3md5oumk3amjk2b14f5qa7sfp77iq&rid=giphy.gif&ct=g"
    />
    <p>
      Prueba
    </p>
  </div>
</div>
`;
